modules/edu.gemini.giapi.client/src/main/scala/edu/gemini/giapi/client/gpi/GPIClient.scala

```diff
@@ -174,7 +184,7 @@ object GPIExample extends App {
     Stream.bracket(
       Giapi.giapiConnection[IO]("failover:(tcp://127.0.0.1:61616)", 2000.millis).connect)(
       giapi => {
-        val client = new GPIClient[IO](giapi)
+        val client = new GPIClient[IO](giapi, scala.concurrent.ExecutionContext.Implicits.global)
```

Comment:
Same here, shouldn't it use the `ExecutionContext` that was passed as a constructor parameter?

Reply:
We should keep an eye on this. We may end up needing a different EC here and elsewhere.

Reply:
we might. note this is an example app